dsn_test.go

```diff
@@ -1,7 +1,6 @@
-package exasol_test
+package exasol
```

Comment:
Why did you change the package?

Reply:
I removed the import and want to access ParseDSN func without reference to `exasol`. Is it wrong?

Reply:
With removing the `_test` we can now access internal code in the tests. Depends with we want to test this too or only the public methods.

Reply:
I think it's fine for now as those are unit tests

Reply:
It's also more consistent, all unit tests belong to `exasol` package now. for some of them I had to use private structures to test